lib/metasploit/framework/obfuscation/crandomizer/random_statements.rb

```diff
@@ -0,0 +1,130 @@
+require 'metasploit/framework/obfuscation/crandomizer/utility'
+
+module Metasploit
+  module Framework
+    module Obfuscation
+      module CRandomizer
+
+        class RandomStatements
+
+          attr_reader :parser
+          attr_reader :fake_function_collection
+          attr_reader :function_list
+
+          # Initializes the RandomStatements class.
+          #
+          # @param fake_functions [Metasploit::Framework::Obfuscation::CRandomizer::CodeFactory::FakeFunctionCollection]
+          # @param s [Metasm::C::Declaration]
+          def initialize(p, fake_functions, s=nil)
+            @parser = p
+            @fake_function_collection = fake_functions
+            @function_list = [ Proc.new { get_random_statements } ]
+
+            # Only generate fake function calls when the function we are modifying isn't
+            # from one of those fake functions (to avoid a recursion).
+            if s && fake_function_collection && !fake_function_collection.has_function_name?(s.var.name)
+              @function_list << Proc.new { get_random_function_call }
+            end
+          end
+
+          # Returns a random statement.
+          #
+          # @return [Array<Metasm::C::CExpression>]
+          # @return [Array<Metasm::C::Declaration>]
```

Comment:
Because sometimes a statement can be an expression, sometimes it can be a declaration. They are all statements that we can use to inject into the source code.